platform/security/session-management-endpoint/src/main/java/org/codice/ddf/security/session/management/impl/SessionManagementEndpointImpl.java

```diff
@@ -0,0 +1,151 @@
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.session.management.impl;
+
+import ddf.security.SecurityConstants;
+import ddf.security.Subject;
+import ddf.security.assertion.SecurityAssertion;
+import ddf.security.assertion.impl.SecurityAssertionImpl;
+import ddf.security.common.SecurityTokenHolder;
+import ddf.security.service.SecurityManager;
+import ddf.security.service.SecurityServiceException;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.time.Clock;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+import javax.ws.rs.core.Response;
+import org.apache.cxf.ws.security.tokenstore.SecurityToken;
+import org.codice.ddf.security.handler.api.SAMLAuthenticationToken;
+import org.codice.ddf.security.session.management.service.SessionManagementCommonService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SessionManagementEndpointImpl implements SessionManagementCommonService {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SessionManagementEndpointImpl.class);
+
+  private SecurityManager securityManager;
+
+  private Clock clock = Clock.systemUTC();
+
+  private String rootContext;
+
+  public InputStream getExpiry(HttpServletRequest request) {
+    HttpSession session = request.getSession(false);
+    long timeLeft = 0;
+    if (session != null) {
+      Object securityToken = session.getAttribute(SecurityConstants.SAML_ASSERTION);
+      if (securityToken instanceof SecurityTokenHolder) {
+        timeLeft = getTimeLeft((SecurityTokenHolder) securityToken);
+      }
+    }
+    return new ByteArrayInputStream(Long.toString(timeLeft).getBytes(StandardCharsets.UTF_8));
+  }
+
+  public InputStream getRenewal(HttpServletRequest request) {
+    HttpSession session = request.getSession(false);
+    Response[] response = new Response[1];
+    InputStream inputStream = null;
+    if (session != null) {
+      Object securityToken = session.getAttribute(SecurityConstants.SAML_ASSERTION);
+      if (securityToken instanceof SecurityTokenHolder) {
+        SecurityTokenHolder tokenHolder = (SecurityTokenHolder) securityToken;
+        Map<String, SecurityToken> realmTokenMap = tokenHolder.getRealmTokenMap();
+        realmTokenMap
+            .keySet()
+            .forEach(
+                s -> {
+                  try {
+                    doRenew(s, realmTokenMap.get(s), tokenHolder);
+                  } catch (SecurityServiceException e) {
+                    response[0] = Response.serverError().build();
+                    LOGGER.error("Failed to renew", e);
+                  }
+                });
+        if (response[0] != null) {
```

Comment:
✏️ Instead of using a Response array, you could just use a boolean that gets set to true when a SecurityServiceException is thrown.